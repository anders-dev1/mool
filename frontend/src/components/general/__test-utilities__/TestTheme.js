export const testTheme = {
    textColor: "black",
    primaryColor: "#007BFF",
    primaryHover: "rgba(0,123,255,0.33)",
    successColor: "#28A745",
    errorColor: "#F02849",
    backgroundColor: "#F0F2F5",
    buttonTextColor: "#FFFFFF",
    dividerColor: "#C7CBD1",
    inputColor: "#99A5B0",
    spinnerColor: "#FFFFFF transparent transparent transparent",
    font: "Acumin Pro, Arial, sans-serif !important;"
};