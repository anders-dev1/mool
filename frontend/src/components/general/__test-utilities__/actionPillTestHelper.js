import {screen} from "@testing-library/react";

export function findActionPillByIcon(icon) {
    return screen.getByText(icon).closest('div');
}

export function findActionPillNumberByIcon(icon) {
    const pill = findActionPillByIcon(icon);
    const numberP = pill.lastChild.lastChild;
    return numberP.textContent;
}