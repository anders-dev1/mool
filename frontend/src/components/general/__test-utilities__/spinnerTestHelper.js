export function findSpinner(container){
    return container.getElementsByClassName('lds-ring')[0];
}