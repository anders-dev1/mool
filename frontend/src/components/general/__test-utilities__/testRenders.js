import React from 'react'
import {render} from '@testing-library/react'
import {Provider} from 'react-redux'
import {setupStore, store} from "../../../redux/store";
import {testTheme} from "./TestTheme";
import {ThemeProvider} from "styled-components";

// This is used to set up the store for tests.
export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        testStore = setupStore(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({children}) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return {testStore, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}

// This is used with the react-test-renderer to render dom in tests.
export function internalsForRenderer(
    ui,
    {
        preloadedState = {},
        testStore = setupStore(preloadedState)
    } = {}
){
    function Wrapper(children) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return Wrapper(ui);
}