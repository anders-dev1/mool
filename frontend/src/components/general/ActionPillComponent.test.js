import renderer from 'react-test-renderer'
import ActionPill from "./ActionPillComponent";
import {fireEvent, render} from "@testing-library/react";
import {ThemeProvider} from "styled-components";
import {testTheme} from "./__test-utilities__/TestTheme";

test('renders correctly', () => {
    const tree = renderer
        .create(
            <ThemeProvider theme={testTheme}>
                <ActionPill className={"test"} link={"test.com"} color={"blue"} number={42} icon={"❤"}/>
            </ThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("triggers passed onClick event when clicked", () => {
    // Arrange
    const mockCallBack = jest.fn();
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <ActionPill onClick={mockCallBack}/>
        </ThemeProvider>);
    const component = container.querySelector('div');

    // Act
    fireEvent.click(component);

    // Assert
    expect(mockCallBack).toHaveBeenCalled();
});

test("cursor becomes pointer on hover", () => {
    // Arrange
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <ActionPill/>
        </ThemeProvider>);
    const component = container.querySelector('div');

    // Assert
    // The only way i know how to test this right now.
    expect(component).toHaveStyle("cursor: pointer;");
});

test("default color should be primary in theme", () => {
    // Arrange
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <ActionPill/>
        </ThemeProvider>);

    // Assert
    const component = container.querySelector('div');
    expect(component).toHaveStyle(`background-color: ${testTheme.primaryColor};`);
});

test("should use the given color", () => {
   // Arrange
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <ActionPill color={"red"}/>
        </ThemeProvider>);

    // Assert
    const component = container.querySelector('div');
    expect(component).toHaveStyle(`background-color: red;`);
});

test("should show a primary color spinner when provided isProcessing is true", () => {
    // Arrange
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <ActionPill processing={true}/>
        </ThemeProvider>
    )

    // Assert
    const spinnerContainerDiv = container.getElementsByClassName('lds-ring')[0];
    const spinnerPiece = spinnerContainerDiv.childNodes[0];
    expect(spinnerPiece).toHaveStyle(`border-color: ${testTheme.primaryColor} transparent transparent transparent`);
});