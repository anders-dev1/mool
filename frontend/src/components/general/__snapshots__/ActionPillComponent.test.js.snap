// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-dkzDqf kwjwjD test"
  color="blue"
  href="test.com"
>
  <p
    className="sc-hKMtZM hepDzL"
  >
    ‚ù§
  </p>
  <div
    className="sc-eCYdqJ cqBlsq"
  >
    <p
      className="sc-jSMfEi fJbvrv"
      data-testid="number"
    >
      42
    </p>
  </div>
</div>
`;
